neuron build on OS X without --disable-pysetup

Hi Michael,

I have been using `--disable-pysetup`. If I remove that, on OS X we get error like:

```
	fi
 /usr/local/bin/ginstall -c -m 644 nrnconf.h '/Users/travis/build/pramodskumbhar/spack-packages/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/neuron-develop-gmfle/x86_64/lib'
 /usr/local/bin/gmkdir -p '/Users/travis/build/pramodskumbhar/spack-packages/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/neuron-develop-gmfle/share/nrn'
 /usr/local/bin/ginstall -c libtool '/Users/travis/build/pramodskumbhar/spack-packages/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/neuron-develop-gmfle/share/nrn'
running install
error: must supply either home or prefix/exec-prefix -- not both
make[3]: *** [install-exec-hook] Error 1
make[2]: *** [install-exec-am] Error 2
make[1]: *** [install-am] Error 2
make: *** [install-recursive] Error 1
``` 

If we scroll to the end of build log [here](https://s3.amazonaws.com/archive.travis-ci.org/jobs/279562105/log.txt?X-Amz-Expires=30&X-Amz-Date=20170925T163530Z&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAJRYRXRSVGNKPKO5A/20170925/us-east-1/s3/aws4_request&X-Amz-SignedHeaders=host&X-Amz-Signature=9a72ba84c77a1965bef7023706ae589c6152b90f763c8e6c50429f153761027b), we can see this error. Is this something relevant? 

This builds fine on linux.

When there is no --disable-pysetup or no --pysetup=....
the default is to install the neuron module in <prefix>/lib/python
ie. in nrn/Makefile.am
$(PYTHON) setup.py install @ac_pysetup@
where in nrn/m4/nrnpython.m4

ac_pysetup='--home=$(prefix)'

So it is not clear to my why it would complain about
error: must supply either home or prefix/exec-prefix -- not both

Please look at <nrnbuild>/config.status and look at the value of (on my linux machine)
S["ac_pysetup"]="--home=$(prefix)"
________________________________________
From: Pramod S Kumbhar <notifications@github.com>
Sent: Monday, September 25, 2017 12:36 PM
To: nrnhines/nrn
Cc: Subscribed
Subject: [nrnhines/nrn] neuron build on OS X without --disable-pysetup (#36)

Hi Michael,

I have been using --disable-pysetup. If I remove that, on OS X we get error like:

        fi
 /usr/local/bin/ginstall -c -m 644 nrnconf.h '/Users/travis/build/pramodskumbhar/spack-packages/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/neuron-develop-gmfle/x86_64/lib'
 /usr/local/bin/gmkdir -p '/Users/travis/build/pramodskumbhar/spack-packages/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/neuron-develop-gmfle/share/nrn'
 /usr/local/bin/ginstall -c libtool '/Users/travis/build/pramodskumbhar/spack-packages/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/neuron-develop-gmfle/share/nrn'
running install
error: must supply either home or prefix/exec-prefix -- not both
make[3]: *** [install-exec-hook] Error 1
make[2]: *** [install-exec-am] Error 2
make[1]: *** [install-am] Error 2
make: *** [install-recursive] Error 1


If we scroll to the end of build log here<https://urldefense.proofpoint.com/v2/url?u=https-3A__s3.amazonaws.com_archive.travis-2Dci.org_jobs_279562105_log.txt-3FX-2DAmz-2DExpires-3D30-26X-2DAmz-2DDate-3D20170925T163530Z-26X-2DAmz-2DAlgorithm-3DAWS4-2DHMAC-2DSHA256-26X-2DAmz-2DCredential-3DAKIAJRYRXRSVGNKPKO5A_20170925_us-2Deast-2D1_s3_aws4-5Frequest-26X-2DAmz-2DSignedHeaders-3Dhost-26X-2DAmz-2DSignature-3D9a72ba84c77a1965bef7023706ae589c6152b90f763c8e6c50429f153761027b&d=DwMCaQ&c=cjytLXgP8ixuoHflwc-poQ&r=RYxvh9hpXg4aGcXdc7VKN3LhQcIm2Bs9VRtAs_7qCcE&m=c5QJYhaunVrvghGPereyGIE7EJ_Bw4BHlsqollnR-T8&s=h0ww_7xPnhS3Tslx-vPsh5_Lr4kgBKz91gGQTxCA6No&e=>, we can see this error. Is this something relevant?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_nrnhines_nrn_issues_36&d=DwMCaQ&c=cjytLXgP8ixuoHflwc-poQ&r=RYxvh9hpXg4aGcXdc7VKN3LhQcIm2Bs9VRtAs_7qCcE&m=c5QJYhaunVrvghGPereyGIE7EJ_Bw4BHlsqollnR-T8&s=vohn6rJpghSr3nLJX27y4lmM_BnaIvGDgrWDt6eCVMo&e=>, or mute the thread<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AEMS0aDN6JaQHpGLyaxlxx-2DfqUV7DyCaks5sl9amgaJpZM4PjAyp&d=DwMCaQ&c=cjytLXgP8ixuoHflwc-poQ&r=RYxvh9hpXg4aGcXdc7VKN3LhQcIm2Bs9VRtAs_7qCcE&m=c5QJYhaunVrvghGPereyGIE7EJ_Bw4BHlsqollnR-T8&s=S8tRRwHcsrLCrU631TQpwWbhwD3wYQDl7qfufH4tQSg&e=>.

--- a/.travis/install_dependencies.sh
+++ b/.travis/install_dependencies.sh
@@ -10,6 +10,12 @@ case "$TRAVIS_OS_NAME" in
         brew install flex bison modules
         brew install python3
 
+        # NOTE: brew installed python3 on OSX has known issue
+        # See https://stackoverflow.com/questions/24257803/
+        rm -f ~/.pydistutils.cfg
+        echo "[install]" > ~/.pydistutils.cfg
+        echo "prefix=" >> ~/.pydistutils.cfg
+
         case "$MPI_LIB_NAME" in
             mpich|mpich3)
                 brew install mpich

--- a/.travis/install_neuron.sh
+++ b/.travis/install_neuron.sh
@@ -6,7 +6,7 @@ if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
 
     if [ "$TRAVIS_OS_NAME" == "linux" ]; then
         packages=(
-                "neuron@develop +python +mpi -shared %gcc ^python@2.7"
+                "neuron@develop +python +mpi +shared %gcc ^python@2.7"
                 "neuron@develop +python +mpi -shared %gcc ^python@3"
                 "neuron@develop -python -mpi -shared %gcc"
                 "neuron@develop +python +mpi +shared %gcc"
@@ -15,7 +15,7 @@ if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
         )
     else
         packages=(
-                "neuron@develop +python +mpi -shared %clang ^python@2.7"
+                "neuron@develop +python +mpi +shared %clang ^python@2.7"
                 "neuron@develop -python -mpi -shared %clang"
                 "neuron@develop +python +mpi +shared %clang ^python@3"
                 "neuron@develop +python +mpi +shared %gcc ^python@2.7"

